service/users_test.go

```diff
@@ -0,0 +1,379 @@
+package service
+
+import (
+	domain "example.com/m/v2/domain"
+	mock_domain "example.com/m/v2/domain/mocks"
+	"fmt"
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var testErr = errors.New("test error")
+
+func TestAddUser(t *testing.T) {
+	testTable := []struct {
+		name                 string
+		inputUser            domain.User
+		returnUser           *domain.User
+		returnError          error
+		callRepositoryMethod bool
+		returnErrorRepo      error
+		checkError           bool
+	}{
+		{
+			name: "ok",
+			inputUser: domain.User{
+				Login:    "testingLogin",
+				Surname:  "testingSurname",
+				Name:     "testungName",
+				Role:     1,
+				Password: "qeqwe12",
+			},
+			returnUser: &domain.User{
+				ID:       1,
+				Login:    "testingLogin",
+				Surname:  "testingSurname",
+				Name:     "testungName",
+				Role:     1,
+				Password: "qeqwe12",
+			},
+			returnError:          nil,
+			callRepositoryMethod: true,
+			returnErrorRepo:      nil,
+			checkError:           false,
+		},
+		{
+			name: "no login",
+			inputUser: domain.User{
+				Login:    "",
+				Surname:  "testingSurname",
+				Name:     "testungName",
+				Role:     1,
+				Password: "qeqwe12",
+			},
+			returnUser:           nil,
+			returnError:          domain.ErrNoLogin,
+			callRepositoryMethod: false,
+			returnErrorRepo:      nil,
+			checkError:           true,
+		},
+		{
+			name: "no surname",
+			inputUser: domain.User{
+				Login:    "tetetet",
+				Surname:  "",
+				Name:     "testungName",
+				Role:     1,
+				Password: "qeqwe12",
+			},
+			returnUser:           nil,
+			returnError:          domain.ErrNoSurname,
+			callRepositoryMethod: false,
+			returnErrorRepo:      nil,
+			checkError:           true,
+		},
+	}
+
+	for _, testCase := range testTable {
+		t.Run(testCase.name, func(t *testing.T) {
+			c := gomock.NewController(t)
+			mockUsersRepository := mock_domain.NewMockUsersStorage(c)
+			usersService := NewUserService(mockUsersRepository)
+
+			if testCase.callRepositoryMethod {
+				mockUsersRepository.EXPECT().Add(testCase.inputUser).Return(testCase.returnUser, testCase.returnErrorRepo)
```

Comment:
Я обычно делаю так. Так проще читается
```suggestion
				mockUsersRepository.
				EXPECT().
				Add(testCase.inputUser).
				Return(testCase.returnUser, testCase.returnErrorRepo)
```

Reply:
Сделаны переносы строк